src/domain/events/EventsPage.tsx

```diff
@@ -139,50 +141,67 @@ const EventsPage: React.FC = () => {
               </div>
             </div>
           </div>
-          <LoadingSpinner isLoading={loading}>
+          <LoadingSpinner isLoading={loadingEvents}>
             {!!eventsWithComingOccurrences.length ? (
-              <Events
-                events={eventsWithComingOccurrences}
-                goToEventOccurrencesPage={goToEventOccurrencesPage}
-              />
+              <>
+                <Events
+                  events={eventsWithComingOccurrences}
+                  goToEventOccurrencesPage={goToEventOccurrencesPage}
+                />
+                {eventsHasNextPage && (
+                  <ShowMoreButton
+                    loading={isLoadingMoreEvents}
+                    onClick={fetchMoreEvents}
+                  />
+                )}
+              </>
             ) : (
               <p>{t('events.textNoComingEvents')}</p>
             )}
 
             {!!eventsWithoutOccurrences.length && (
               <>
                 <EventsTitle
-                  count={eventsWithoutOccurrences.length}
+                  count={
+                    eventsWithoutOccurrencesCount ||
+                    eventsWithoutOccurrences.length
+                  }
                   title={t('events.titleEventsWithoutOccurrences')}
                 />
                 <Events
                   events={eventsWithoutOccurrences}
                   goToEventOccurrencesPage={goToEventOccurrencesPage}
                 />
+                {eventsWithoutOccurrencesHasNextPage && (
+                  <ShowMoreButton
+                    loading={loadingMoreEventsWithoutOccurrences}
+                    onClick={fetchMoreEventsWithoutOccurrences}
+                  />
+                )}
               </>
             )}
+
             {!!eventsWithPastOccurrences.length && (
               <>
                 <EventsTitle
-                  count={eventsWithPastOccurrences.length}
+                  count={
+                    eventsWithPastOccurrencesCount ||
+                    eventsWithPastOccurrences.length
+                  }
                   title={t('events.titleEventsWithPastOccurrences')}
                 />
                 <Events
                   events={eventsWithPastOccurrences}
                   goToEventOccurrencesPage={goToEventOccurrencesPage}
                 />
+                {pastEventsHasNextPage && (
+                  <ShowMoreButton
+                    loading={loadingMorePastEvents}
+                    onClick={fetchMorePastEvents}
+                  />
+                )}
               </>
             )}
```

Comment:
Maybe you could have separate component to show each event list to avoid duplicate code

Reply:
Thought about that but I don't know if there is a huge benefit because the duplication happens only twice here (events with upcoming occurrences has little bit different structure). And adding new abstraction level (new component) might make it little bit harder to understand.

Don't know... maybe it would be little bit easier to read. I will add new component :)

Reply:
I undestand. I almost didn't even comment about it. 

It would be nice to move event fetch hooks inside that component. But I know the loading state is the reason to keep them on this level

Reply:
Hmm also moving fetching logic inside a event category component crossed my mind but I can't remember why I thought it would not be good idea.

Now that I look the code again I feel like there is no reason not to make a new component that has the fetching logic inside it, just pass query variables as prop? 🤔

Reply:
Yeah, passing query variables and title should be enough

Reply:
I was just thinking that maybe that component could be used for the upcoming occurrences (just don't pass title prop). But that is when the loading state becomes issue.

Maybe I have to ask the designer how we want to handle the loading states in the first place :)